draco/analysis/beamform.py

```diff
@@ -0,0 +1,515 @@
+import numpy as np
+from caput import mpiarray, config, mpiutil, pipeline
+from ..core import task, containers, io
+from cora.util import units
+from ch_util import tools, ephemeris
+from drift.telescope import cylbeam
+from ..util._fast_tools import beamform
+from ..util.tools import polarization_map, baseline_vector
+from mpi4py import MPI
+
+# Constants
+NU21 = units.nu21
+C = units.c
+
+
+class BeamFormBase(task.SingleTask):
+    """ Base class for beam forming tasks.
+        Defines a few useful methods. Not to be used directly
+        but as parent class for BeamForm and BeamFormCat.
+
+    Attributes
+    ----------
+    collapse_ha : bool
+        Wether or not to sum over hour-angle/time to complete
+        the beamforming. Default is True, which sums over.
+    polarization : string
+        One of:
+        'I' : Stokes I only.
+        'full' : 'XX', 'YY', 'XY' and 'YX' in this order.
```

Comment:
The polarisation order should be `XX`, `XY`, `YX`, `YY` to be consistent with everywhere else that we use this (e.g. ring maps)